command/resource/helper.go

```diff
@@ -95,17 +100,41 @@ func ParseResourceFromFile(filePath string) (*pbresource.Resource, error) {
 		return nil, fmt.Errorf("Failed to load data: %v", err)
 	}
 	var parsedResource *pbresource.Resource
-	parsedResource, err = resourcehcl.Unmarshal([]byte(data), consul.NewTypeRegistry())
-	if err != nil {
+	if isHCL([]byte(data)) {
+		parsedResource, err = resourcehcl.Unmarshal([]byte(data), consul.NewTypeRegistry())
+	} else {
 		parsedResource, err = parseJson(data)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to decode resource from input file: %v", err)
-		}
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decode resource from input file: %v", err)
 	}
 
 	return parsedResource, nil
 }
 
+// this is an inlined variant of hcl.lexMode()
+func isHCL(v []byte) bool {
```

Comment:
This was lifted from the call stack for `consul config write` which needs to do something similar: https://github.com/hashicorp/consul/blob/a9747dc38c7c916f6834b259a7cc375886a0ff0a/command/helpers/decode_shim.go#L91